griptape/memory/structure/conversation_memory.py

```diff
@@ -17,17 +18,60 @@ class ConversationMemory:
     runs: list[Run] = field(factory=list, kw_only=True)
     structure: Structure = field(init=False)
     autoload: bool = field(default=True, kw_only=True)
+    autoprune: bool = field(default=True, kw_only=True)
 
     def __attrs_post_init__(self) -> None:
         if self.driver and self.autoload:
             memory = self.driver.load()
             if memory is not None:
                 [self.add_run(r) for r in memory.runs]
 
-    def add_to_prompt_stack(self, stack: PromptStack) -> None:
-        for r in self.runs:
-            stack.add_user_input(r.input)
-            stack.add_assistant_input(r.output)
+    def add_to_prompt_stack(self, stack: PromptStack, index: Optional[int]=None) -> None:
```

Comment:
I'm wondering if this should live in `PromptStack.add_memory(memory, index)` instead. This way the object (prompt stack) is modifying itself instead of another entity (memory) modifying the object. The former is generally less bug-prone. Let me know if that breaks anything though.

Reply:
I agree, and it does make for a nicer interface. Though it gets kind of ugly with other memory types like `SummaryConversationMemory` which adds `summary` and `unsummarized_runs` to the stack.

And typing that comment made me realize I did not implement pruning logic for `SummaryConversationMemory` 😅

Reply:
Made this refactor, it's much cleaner.